### PR TITLE
TrackPageView refactoring

### PR DESCRIPTION
**Pull request by MaxShehovtsov**
Issue #94

### CODE REVIEW

**Comment by MaxShehovtsov** [comment]
ping - @AlexBulankou @madeleinefletcher @kamilszostak

File: JavaScript/JavaScriptSDK/Telemetry/PageViewManager.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            appInsights: IAppInsightsInternal,
+            overridePageViewDuration: boolean) {
+            this.overridePageViewDuration = overridePageViewDuration;
+            this.appInsights = appInsights;
+        }
+
+        /**
+        * Currently supported cases:
+        * 1) (default case) track page view called with default parameters, overridePageViewDuration = false. Page view is sent with page view performance when navigation timing data is available.
+        *    If navigation timieng is not supported then page view is sent right away with 0 duration. Page view performance is not sent.
```
Comment (kamilszostak) [inline JavaScript/JavaScriptSDK/Telemetry/PageViewManager.ts]: Typo: navigation timing

File: JavaScript/JavaScriptSDK/Telemetry/PageViewManager.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                        var pageViewPerformance = new Telemetry.PageViewPerformance(name, url, null, properties, measurements);
+
+                        if (!pageViewPerformance.getIsValid() && !pageViewSent) {
+                            // If navigation timing gives invalid numbers, then go back to "override page view duration" mode.
+                            // That's the best value we can get that makes sense.
+                            this.appInsights.sendPageViewInternal(name, url, customDuration, properties, measurements);
+                            this.appInsights.flush();
+                        } else {
+                            if (!pageViewSent) {
+                                this.appInsights.sendPageViewInternal(name, url, pageViewPerformance.getDurationMs(), properties, measurements);
```
Comment (kamilszostak) [inline JavaScript/JavaScriptSDK/Telemetry/PageViewManager.ts]: Don't you need to set pageViewSent = true after you send the PageView trace?

File: JavaScript/JavaScriptSDK/Telemetry/PageViewManager.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                            }
+
+                            if (!this.pageViewPerformanceSent) {
```
Comment (kamilszostak) [inline JavaScript/JavaScriptSDK/Telemetry/PageViewManager.ts]: Do we have a unit test for this chase?

File: JavaScript/JavaScriptSDK/Telemetry/PageViewManager.ts (hunk 2 above)
Comment (MaxShehovtsov) [inline JavaScript/JavaScriptSDK/Telemetry/PageViewManager.ts]: I could, but its unnecessary - timeout is alreday cleared, and noone down below tries to send it (without entering the timeout again)

**Pull request merged**